datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala

```diff
@@ -127,6 +136,45 @@ object DatasetComparisonJob {
     }
   }
 
+  private def handleDuplicates(outPath: String,
+                               dfsWithKey: GenericPair[DataFrame]): (GenericPair[Long], GenericPair[DataFrame]) = {
+    def writeDown(df: DataFrame, duplicates: DataFrame, path: String): Unit = {
+      df.alias("original")
+        .join(duplicates, Seq(comparisonUniqueId), "left_outer")
+        .select("original.*")
+        .write
+        .format("parquet")
+        .save(path)
+    }
+
+    val dfsDuplicates: GenericPair[Option[DataFrame]] = GenericPair(
+      checkForDuplicateRows(dfsWithKey.reference),
+      checkForDuplicateRows(dfsWithKey.actual)
+    )
+
+    val duplicateCounts: GenericPair[Long] = GenericPair(
+      dfsDuplicates.reference.map(_.count()).getOrElse(0),
+      dfsDuplicates.actual.map(_.count()).getOrElse(0)
+    )
+
+    if ((duplicateCounts.reference + duplicateCounts.actual) > 0 && !allowDuplicates) {
+      dfsDuplicates.reference.foreach(x => writeDown(dfsWithKey.actual, x, s"$outPath/newDuplicates"))
+      dfsDuplicates.actual.foreach(x => writeDown(dfsWithKey.actual, x, s"$outPath/newDuplicates"))
```

Comment:
Don't these two calls to `writeDown` cause an error? Seems they both end up trying to write to the same path.

Reply:
This is a copypaste error. Will fix.